pkg/asset/machines/gcp/zones.go

```diff
@@ -5,18 +5,23 @@ import (
 	"fmt"
 	"time"
 
+	"google.golang.org/api/option"
 	"k8s.io/apimachinery/pkg/util/sets"
 
-	configv1 "github.com/openshift/api/config/v1"
 	gcpconfig "github.com/openshift/installer/pkg/asset/installconfig/gcp"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
 )
 
 // ZonesForInstanceType retrieves a filtered list of availability zones where
 // the particular instance type is available. This is mainly necessary for
 // arm64, since the instance t2a-standard-* is not available in all
 // availability zones.
-func ZonesForInstanceType(project, region, instanceType string, serviceEndpoints []configv1.GCPServiceEndpoint) ([]string, error) {
-	svc, err := gcpconfig.GetComputeService(context.Background(), serviceEndpoints)
+func ZonesForInstanceType(project, region, instanceType string, endpoint *gcptypes.PSCEndpoint) ([]string, error) {
+	opts := []option.ClientOption{}
+	if endpoint != nil && endpoint.ClusterUseOnly != nil && !(*endpoint.ClusterUseOnly) {
```

Comment:
There are many instances of this check. Should we extract it into a common function? For example:

```golang
// shouldApplyEndpointToInstaller returns true if the PSC endpoint should be used
// during installation (not just by the cluster).
func shouldApplyEndpointToInstaller(endpoint *gcptypes.PSCEndpoint) bool {
}
```